cmake/CMakeLists.txt

```diff
@@ -718,25 +725,42 @@ if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 endif()
 
 #Dependencies begin
+message("Loading Dependencies URLs ...")
+
+file(STRINGS deps.txt ONNXRUNTIME_DEPS_LIST)
+foreach(ONNXRUNTIME_DEP IN LISTS ONNXRUNTIME_DEPS_LIST)
+  list(POP_FRONT ONNXRUNTIME_DEP ONNXRUNTIME_DEP_NAME)
+  list(POP_FRONT ONNXRUNTIME_DEP ONNXRUNTIME_DEP_URL)
+  set(DEP_URL_${ONNXRUNTIME_DEP_NAME} ${ONNXRUNTIME_DEP_URL})
+  set(DEP_SHA1_${ONNXRUNTIME_DEP_NAME} ${ONNXRUNTIME_DEP})
+  message("${DEP_URL_${ONNXRUNTIME_DEP_NAME}}")
+endforeach()
+
+message("Loading Dependencies ...")
+
 if (onnxruntime_BUILD_UNIT_TESTS)
   if (onnxruntime_PREFER_SYSTEM_LIB)
     find_package(GTest CONFIG)
   endif()
   if (TARGET GTest::gtest)
     message("Use gtest from preinstalled system lib")
   else()
-    message("Use gtest from submodule")
-
     # WebAssembly threading support in Node.js is still an experimental feature and
     # not working properly with googletest suite.
     if (onnxruntime_BUILD_WEBASSEMBLY)
       set(gtest_disable_pthreads ON)
     endif()
 
     # gtest and gmock
-    set_msvc_c_cpp_compiler_warning_level(4)
-    add_subdirectory(${PROJECT_SOURCE_DIR}/external/googletest EXCLUDE_FROM_ALL)
-    set_msvc_c_cpp_compiler_warning_level(3)
+
+    FetchContent_Declare(
+      googletest
+      URL ${DEP_URL_googletest}
+      FIND_PACKAGE_ARGS NAMES GTest
+      URL_HASH SHA1=${DEP_SHA1_googletest}
+    )
+    FetchContent_MakeAvailable(googletest)
```

Comment:
This is a really good suggestion. I think the key point is that if a dependency of onnxruntime, for example ort-extension, also needs to use a same component that ORT uses,  then ORT should declaring the dependency first. For example, ORT should declare absl before including ORT-extension.  At this moment, we can manually adjust the order. I don't like to have a bigger change in this PR.  If we move all FetchContent_Declare in one place, then we don't need to have the wil.cmake and the similar ones.  We will need to structure the code in a different way since FetchContent_Declare/FetchContent_MakeAvailable happens in two stages.  I don't favor to do it now.

Reply:
it's fine to do it in a later PR